crates/core/src/sql/execute.rs

```diff
@@ -187,31 +191,47 @@ pub struct SqlResult {
 
 /// Run the `SQL` string using the `auth` credentials
 pub async fn run(
-    db: &RelationalDB,
-    sql_text: &str,
+    db: Arc<RelationalDB>,
+    sql_text: String,
     auth: AuthCtx,
-    subs: Option<&ModuleSubscriptions>,
-    module: Option<&ModuleHost>,
+    subs: Option<ModuleSubscriptions>,
+    module: Option<ModuleHost>,
     head: &mut Vec<(Box<str>, AlgebraicType)>,
 ) -> Result<SqlResult, DBError> {
+    match module {
+        Some(module) => {
+            let info = module.info.clone();
+            module.call_view_sql(info, db, sql_text, auth, subs).await
+        }
+        None => run_from_module::<crate::host::wasmtime::WasmtimeInstance>(None, db, sql_text, auth, subs, head)
+            .map(|x| x.0),
+    }
```

Comment:
Could you add some commentary about when `module` will be `Some` or `None`, and what each of these cases means for SQL execution? Reading this code, going from "there is no module" to `run_in_module` feels like nonsense.

Reply:
I promise you to raise next PR to add commentary and make code more readable, but will prefer to merge this one right now so I don't want to wait for CIs again :)